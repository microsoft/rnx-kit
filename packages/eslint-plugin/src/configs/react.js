// @ts-check
"use strict";

const { FlatCompat } = require("@eslint/eslintrc");
const react = require("eslint-plugin-react");
const eslint = require("./eslint");

const compat = new FlatCompat({
  baseDirectory: __dirname,
  recommendedConfig: eslint.configs.recommended,
});

module.exports = [
  ...compat.extends("plugin:react-hooks/recommended"),
  react.configs.flat.recommended,
  {
    name: "rnx-kit/react",
    rules: {
      "react/prop-types": "off",
    },
    settings: {
      react: {
        version: "detect",
      },
    },
  },
];
