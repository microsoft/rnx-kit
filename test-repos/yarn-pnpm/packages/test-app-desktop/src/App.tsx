import type * as React from "react";

export const App = (): React.ReactElement | null => null;
