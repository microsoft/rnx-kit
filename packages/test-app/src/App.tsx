import * as React from "react";

export default (): React.ReactElement | null => null;
