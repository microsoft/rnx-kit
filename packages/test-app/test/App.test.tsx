import * as React from "react";
import TestRenderer from "react-test-renderer";
import App from "../src/App";

it("renders correctly", () => {
  TestRenderer.create(<App />);
});
