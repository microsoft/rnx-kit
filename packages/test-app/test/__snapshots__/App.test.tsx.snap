// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  style={
    {
      "backgroundColor": "#F3F3F3",
      "flex": 1,
    }
  }
>
  <RCTScrollView
    contentInsetAdjustmentBehavior="automatic"
    onLayout={[Function]}
    style={
      {
        "backgroundColor": "#F3F3F3",
        "flex": 1,
      }
    }
  >
    <View>
      <View
        accessibilityIgnoresInvertColors={true}
        style={
          [
            {
              "paddingBottom": 40,
              "paddingHorizontal": 32,
              "paddingTop": 96,
            },
            {
              "backgroundColor": "#F3F3F3",
            },
          ]
        }
      >
        <Image
          accessibilityRole="image"
          source={
            {
              "testUri": "../Libraries/NewAppScreen/components/logo.png",
            }
          }
          style={
            [
              {
                "bottom": 0,
                "left": 0,
                "position": "absolute",
                "right": 0,
                "top": 0,
              },
              {
                "height": undefined,
                "width": undefined,
              },
              {
                "marginBottom": -192,
                "marginLeft": -128,
                "opacity": 0.2,
                "overflow": "visible",
                "resizeMode": "cover",
              },
            ]
          }
          testID="new-app-screen-header"
        />
        <Text
          style={
            [
              {
                "fontSize": 40,
                "fontWeight": "700",
                "textAlign": "center",
              },
              {
                "color": "#000",
              },
            ]
          }
        >
          Welcome to
          

          React Native
        </Text>
      </View>
      <View
        style={
          {
            "backgroundColor": "#FFF",
            "borderRadius": 8,
            "margin": 16,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 18,
                "marginVertical": 12,
              }
            }
          >
            window.localStorage
          </Text>
          <Text
            style={
              {
                "color": "#444",
                "fontSize": 18,
                "textAlign": "right",
              }
            }
            testID="window.localstorage-value"
          >
            Error
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#DAE1E7",
              "height": 0.5,
              "marginStart": 16,
            }
          }
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 18,
                "marginVertical": 12,
              }
            }
          >
            Acquire Token
          </Text>
        </View>
      </View>
      <View
        style={
          {
            "backgroundColor": "#FFF",
            "borderRadius": 8,
            "margin": 16,
          }
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 18,
                "marginVertical": 12,
              }
            }
          >
            React Native
          </Text>
          <Text
            style={
              {
                "color": "#444",
                "fontSize": 18,
                "textAlign": "right",
              }
            }
            testID="react-native-value"
          >
            0.73.5
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#DAE1E7",
              "height": 0.5,
              "marginStart": 16,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 18,
                "marginVertical": 12,
              }
            }
          >
            Hermes
          </Text>
          <Text
            style={
              {
                "color": "#444",
                "fontSize": 18,
                "textAlign": "right",
              }
            }
            testID="hermes-value"
          >
            Off
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#DAE1E7",
              "height": 0.5,
              "marginStart": 16,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 18,
                "marginVertical": 12,
              }
            }
          >
            Fabric
          </Text>
          <Text
            style={
              {
                "color": "#444",
                "fontSize": 18,
                "textAlign": "right",
              }
            }
            testID="fabric-value"
          >
            Off
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#DAE1E7",
              "height": 0.5,
              "marginStart": 16,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 18,
                "marginVertical": 12,
              }
            }
          >
            Concurrent React
          </Text>
          <Text
            style={
              {
                "color": "#444",
                "fontSize": 18,
                "textAlign": "right",
              }
            }
            testID="concurrent-react-value"
          >
            Off
          </Text>
        </View>
        <View
          style={
            {
              "backgroundColor": "#DAE1E7",
              "height": 0.5,
              "marginStart": 16,
            }
          }
        />
        <View
          style={
            {
              "alignItems": "center",
              "flexDirection": "row",
              "paddingHorizontal": 16,
            }
          }
        >
          <Text
            style={
              {
                "color": "#000",
                "flex": 1,
                "fontSize": 18,
                "marginVertical": 12,
              }
            }
          >
            Bridgeless
          </Text>
          <Text
            style={
              {
                "color": "#444",
                "fontSize": 18,
                "textAlign": "right",
              }
            }
            testID="bridgeless-value"
          >
            Off
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</RCTSafeAreaView>
`;
